visionipc/visionipc_server.cc

```diff
@@ -0,0 +1,175 @@
+#include <iostream>
+#include <chrono>
+#include <cassert>
+#include <random>
+
+#include <poll.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "messaging.hpp"
+// TODO: rename header files to hpp for consistency
+#include "ipc.h"
+#include "visionipc_server.h"
+
+VisionIpcServer::VisionIpcServer(std::string name, cl_device_id device_id, cl_context ctx) : name(name), device_id(device_id), ctx(ctx) {
+  init();
+}
+
+VisionIpcServer::VisionIpcServer(std::string name) : name(name) {
+  init();
+}
+
+void VisionIpcServer::init(void){
+  msg_ctx = Context::create();
+
+  std::random_device rd("/dev/urandom");
+  std::uniform_int_distribution<uint64_t> distribution(0,std::numeric_limits<uint64_t>::max());
+  server_id = distribution(rd);
```

Comment:
can't we use the PID or do we want to support multiple servers in a process?

Reply:
I'd like to support multiple servers in one process